### PR TITLE
Update main.py

### PR DESCRIPTION
**Pull request by PhysikerErlangen**
I got confused by the use of the binary cross entropy. In particular it wasn't clear to me why the variable real_labels are used in the training of the generator.

I have added some comments. I am not sure if they are correct, so you might want to double check them.

### CODE REVIEW

**Comment by yunjey** [comment]
@PhysikerErlangen Thanks. I edited some comments.

**Comment by PhysikerErlangen** [comment]
Hey,
thanks for editing my comments.

I still think you should elaborate in one or two sentences, why we maximize log(D(G(z)) by computation of
`g_loss = criterion(outputs, real_labels)` 
 
You could add: Remark, second term of the loss is always zero since real_labels == 1, thus only log(D(G(z)) is computed

I don't know about others, but this has really confused me for a while because I didn't understand how the BCE loss is connected to the computation of only log D(G(z)). 

Best,
Florian
